### ISSUE TITLE
Ingress Controller not applying WAF Policy

### ISSUE THREAD

**Issue by Danlewis3** [Azure/application-gateway-kubernetes-ingress#766]
**Describe the bug**
Following documentation here: https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/docs/annotations.md#azure-waf-policy-for-path

Trying to Add a policy to an ingress

**To Reproduce**
Ingress controller below, everything else works as it should can access the UI but policy never gets applied and can't see any logs
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: $(ncnamelower)
  annotations:
    kubernetes.io/ingress.class: azure/application-gateway
    appgw.ingress.kubernetes.io/backend-protocol: "https"
    appgw.ingress.kubernetes.io/backend-path-prefix: "/"
    appgw.ingress.kubernetes.io/waf-policy-for-path: "/subscriptions/c2d12071-f2ec-4947-8022-678f59ac26d9/resourceGroups/UK-WAF-POLICIES/providers/Microsoft.Network/applicationGatewayWebApplicationFirewallPolicies/UKWafPolicy"
spec:
  tls:
  - hosts:
    - $(URL)
    secretName: $(SSL)
  rules:
  - host: $(URL)
    http:
      paths:
      - backend:
          serviceName: $(ncnamelower)
          servicePort: 443

**Comment by Danlewis3** [comment]
@akshaysngupta your best prob as you did the coding for it

**Comment by akshaysngupta** [comment]
@Danlewis3 
TLDR;
To resolve this, Add following in your helm config:
```yaml
appgw:
  waf_listener: true
```
This will assign the WAF policy at the listener level instead of path level.

**Issue:**
When an ingress has a simple rule with path: `/`, then a basic routing rule is created on Application Gateway. In a basic routing rule, when waf-policy annotation is used, WAF policy is supposed to be assigned at the listener.
In current implementation, WAF policy is applied at path level only when a specific path is used (ex: `/app`) as this leads to creation of a Path-based routing rule on App Gateway.

**Bypass**
In order to bypass this missing piece, please use the following in your helm config:
```yaml
appgw:
  waf_listener: true
```
Here is how this key is currently implemented.
https://github.com/Azure/application-gateway-kubernetes-ingress/blob/bd619a175318cbb2cecaf3e966313897571ea675/pkg/appgw/frontend_listeners.go#L120

**Comment by Danlewis3** [comment]
When adding the line above to Helm-config the pod fails to get become online as the readiness health check doesn't work

**Comment by Danlewis3** [comment]
Readiness probe failed: Get http://10.189.8.9:8123/health/ready: dial tcp 10.189.8.9:8123: connect: connection refused
appgw:
    waf_listener: true
    subscriptionId: c2d12071-erdf-4947-8022-678f59ac26d9
    resourceGroup: WAF
    name: AKS
    usePrivateIP: false

**Comment by akshaysngupta** [comment]
@Danlewis3 If AGIC pod is failing to come up, can you check the AGIC pod logs for any visible errors?

**Comment by Danlewis3** [comment]
Readiness probe failed: Get http://10.189.9.152:8123/health/ready: dial tcp 10.189.9.152:8123: connect: connection refused

This is what we get on the pod with the line specified
